packages/core/src/sims/ranged/mch/mch_types.ts

```diff
@@ -0,0 +1,117 @@
+import {Ability, GcdAbility, OgcdAbility, Buff, BuffController} from "@xivgear/core/sims/sim_types";
+import {MchGauge} from "./mch_gauge";
+import {removeSelf} from "@xivgear/core/sims/common/utils";
+
+export type MchAbility = Ability & Readonly<{
+    /** Run if an ability needs to update the Heat gauge */
+    updateHeatGauge?(gauge: MchGauge): void;
+
+    /** Run if an ability needs to update the battery gauge */
+    updateBatteryGauge?(gauge: MchGauge): void;
+
+    /** The heat Gauge cost of the ability */
+    heatGaugeCost?: number;
+
+    /** The battery Gauge cost of the ability */
+    batteryGaugeCost?: number;
+}>
+
+export type MchGcdAbility = GcdAbility & MchAbility;
+
+export type MchOgcdAbility = OgcdAbility & MchAbility;
+
+/** MCH ability that costs heat gauge */
+export type heatGaugeAbility = MchAbility & Readonly<{
+    heatGaugeCost: number;
+}>
+
+/** MCH ability that costs heat gauge */
+export type batteryGaugeAbility = MchAbility & Readonly<{
+    batteryGaugeCost: number;
+}>
+
+/** Represents MCH's Gauge state */
+export type MchGaugeState = {
+    heatGauge: number,
+    batteryGauge: number,
+}
+
+
+/** Represents the extra data for UsedAbility, primarily for consumption in the UI */
+export type MchExtraData = {
+    /** The battery gauge data */
+    gauge: MchGaugeState,
+    hypercharge: number,
+};
+
+export const ReassembleBuff: Buff = {
+    name: "Reassemble",

```

Comment:
```suggestion
    name: "Reassembled",
```
buff name from game data